src/main/java/com/hillel/items_exchange/service/UserService.java

```diff
@@ -56,10 +63,20 @@ public boolean registerNewUser(UserRegistrationDto userRegistrationDto, Role rol
     }
 
     public UserDto update(UserDto newUserDto, User user) throws IllegalOperationException {
-        checkReadOnlyFieldsUpdate(newUserDto, user);
+        User updatedUser = convertDto(newUserDto);
+        Collection<Child> children = extractAll(updatedUser.getChildren(), child -> child.getId() == 0, ArrayList::new);
+        Collection<Phone> phones = extractAll(updatedUser.getPhones(), phone -> phone.getId() == 0, HashSet::new);
+        if (!isNewUser(user) && (!children.isEmpty() || !phones.isEmpty())) {
+            throw new IllegalOperationException(
+                    getExceptionMessageSource("exception.illegal.field.change") + "children or phones");
+        }
+        checkReadOnlyFieldsUpdate(updatedUser, user);
         BeanUtils.copyProperties(newUserDto, user,
                 "id", "created", "updated", "status", "username", "password", "online",
                 "lastOnlineTime", "role", "advertisements", "deals", "phones", "children");
+        user.setUpdated(LocalDate.now());
```

Comment:
Yes, it works fine, but in case when user data is not updated but contains changes of child entities it doesn't trigger CustomDateTimeProvider because actually the parent entity wasn't updated. So if a user adds children/phones his status isn't change to UPDATED and he is able to do that as much as he wants

Reply:
Got it, thanks.